lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -1484,21 +1485,28 @@ const MColor& ProxyRenderDelegate::GetWireframeColor() const { return _wireframe
 
 GfVec3f ProxyRenderDelegate::GetCurveDefaultColor()
 {
-    MDoubleArray curveColorResult;
-    {
-        std::lock_guard<std::mutex> mutexGuard(_mayaCommandEngineMutex);
-        MGlobal::executeCommand(
-            "int $index = `displayColor -q -dormant \"curve\"`; colorIndex -q $index;",
-            curveColorResult);
+    // Enter the mutex and check the cache
+    std::lock_guard<std::mutex> mutexGuard(_mayaCommandEngineMutex);
+    if (_dormantCurveColorCache.second == _frameCounter) {
```

Comment:
I think if we make _dormantCurveColorCache.second a std::atomic<uint64_t> then we could move acquiring the mutex to be after this check to see if we already have already read the value this frame. That would prevent lots of mutex acquires which might be faster.

In the case where the test fails we'd have to acquire the mutex and re-test to avoid reading the value two or more times, but that is okay.

Reply:
The std::atomic<uint64_t> approach may be faster but not necessarily. We can try it if we still need to after this change.